code/modules/mob/living/simple_animal/hostile/megafauna/legion.dm

```diff
@@ -196,6 +196,11 @@ Difficulty: Medium
 	. = ..()
 	if(QDELETED(src))
 		return
+	if(!GLOB.necropolis_gate)
+		return
+	if(GLOB.necropolis_gate.legion_triggered)
+		return
+	GLOB.necropolis_gate.toggle_the_gate(src, TRUE)
```

Comment:
You want to not early return here to prevent the logic below from getting cancelled also. Also, there is no second parameter on toggle_the_gate
```suggestion
	if(GLOB.necropolis_gate && !GLOB.necropolis_gate.open)
		GLOB.necropolis_gate.toggle_the_gate(src)
```